### PR TITLE
Fix some bugs when seeking

### PR DESCRIPTION
**Pull request by jaeyoonlim**
Thank you for your great library.
When I was using this library, I have found some bugs when seeking audio.

**Problems**
1. When seeking started, 'OnStartTrackingTouch' in 'JcPlayerView' immediately call function 'showProgressBar(). So 'OnStopTrackingTouch' can't call function 'dismissProgressBar()' because 'seekBar' is already disabled.

2. 'OnProgressChanged' in 'JcPlayerView' call function 'seekTo' every progress changing even just dragging seek bar.

So I fixed these bugs.

**Solutions**
1. I moved functions 'showProgressBar()' and 'seekTo()' to 'OnStopTrackingTouch()'.
And I added new Listener 'OnSeekComplete'. It calls 'dismissProgressBar()' when seeking completed.

**Here is before & after**
Before
before

After
after

### CODE REVIEW

**Comment by jeancsanchez** [comment]
@jaeyoonlim Great job! I'll review that.

**Comment by jeancsanchez** [comment]
@jaeyoonlim Please, update your branch with the latest version code `master_v2`.

**Comment by jaeyoonlim** [comment]
@jeancsanchez OK, I'll update my branch.

**Comment by jaeyoonlim** [comment]
@jeancsanchez I updated my branch. Please, review this.

File: jcplayer/src/main/java/com/example/jean/jcplayer/view/JcPlayerView.kt [hunk 1]
```diff
@@ -478,14 +480,15 @@ class JcPlayerView : LinearLayout, View.OnClickListener, SeekBar.OnSeekBarChange
...
-            showProgressBar()
-        }
-    }
+    override fun onStartTrackingTouch(seekBar: SeekBar) {}
 
     override fun onStopTrackingTouch(seekBar: SeekBar) {
-        dismissProgressBar()
+        if (jcPlayerManager.currentAudio != null) {
+            jcPlayerManager.let {
+                it.seekTo(seekBar.progress)
```
Comment (jeancsanchez) [inline jcplayer/src/main/java/com/example/jean/jcplayer/view/JcPlayerView.kt]: There is no need to use `let`, because the jcPlayerManager is not nullable

**Comment by jeancsanchez** [comment]
@jaeyoonlim  I am tested it, but it not solves the problem yet :/

File: jcplayer/src/main/java/com/example/jean/jcplayer/view/JcPlayerView.kt (hunk 1 above)
Comment (jaeyoonlim) [reply jcplayer/src/main/java/com/example/jean/jcplayer/view/JcPlayerView.kt]: @jeancsanchez I edited it, I have to study Kotlin😅

**Comment by jaeyoonlim** [comment]
@jeancsanchez I tested it again but, it works pretty well..
I may think you did not edit the file 'sample/build.gradle
Please check the file. (implementation -> api project)

screenshot

**Comment by jeancsanchez** [comment]
@jaeyoonlim When the player is paused and the progress bar is changed, the player's behavior is weird.

**Comment by jaeyoonlim** [comment]
output

@jeancsanchez Are you saying this?

**Comment by jeancsanchez** [comment]
@jaeyoonlim Not exactly. Follow this steps:
1 - Pause the player
2 - Handle the seek bar manually (with the player paused)

**Comment by jaeyoonlim** [comment]
@jeancsanchez Oh, I understand, When user seek music slowly, Seek bar is still moving.
Is it right?

**Comment by jaeyoonlim** [comment]
@jeancsanchez I followed your description and I have a question.
When the player is paused and the progress bar is changed, the player is still paused.
Is it right?

**Comment by jeancsanchez** [comment]
@jaeyoonlim Yes.

**Comment by jaeyoonlim** [comment]
@jeancsanchez Please check my branch. I fixed it.

**Comment by jaeyoonlim** [comment]
@jeancsanchez I fixed it. Please check my branch.

**Comment by peace-shillong** [comment]
@jaeyoonlim I've Tested your branch fix_seeking_bug and saw that when I pause an audio and try to seek the seekbar the player is able to continue playing after sometime from the seek position but now I'm unable to use any of the buttons...
screenshot_20180826-224451
This is what i see after moving the seekbar...

**Comment by jaeyoonlim** [comment]
@peace-shillong Thank you for your feedback.
It's original branch (master_v2) bug.
Please check sample.gradle file.

implementation -> api
image

**Comment by peace-shillong** [comment]
capture
This is my sample.gradle file.
I Tried Running the app on Nougat 7.0 and that was what i saw while playing the song. 

I tried connecting with my Jelly Beans Phone and compile but I got compilation error - something to do with dex files...

**Comment by jaeyoonlim** [comment]
@peace-shillong 
Before you run sample app, You have to edit the file like below
screenshot

Your setting is loading JcPlayer from remote not local.
Please re-check it.

**Comment by peace-shillong** [comment]
@jaeyoonlim Thank you.. it works!
But i do see some red lines in the project MainActivity:

**Pull request closed without merging**